kernel/byterun/coq_interp.c

```diff
@@ -76,6 +76,14 @@ sp is a local copy of the global variable extern_sp. */
 #   define print_lint(i)
 #endif
 
+#define CHECK_STACK(num_args) {                                                \
+if (sp - num_args < coq_stack_threshold) {                                     \
+  coq_sp = sp;                                                                 \
+  realloc_coq_stack(num_args * sizeof(value) + Coq_stack_threshold);           \
```

Comment:
Yes, I was a bit confused about this too. So the pre-existing code of `check_stack:` (see below) was also wrong, right? It was calling realloc_coq_stack with Coq_stack_threshold. We should give more explicit names to Coq_stack_threshold vs coq_stack_threshold to avoid confusion.